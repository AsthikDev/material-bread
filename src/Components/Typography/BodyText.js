import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { StyleSheet } from 'react-native';
import BaseText from './BaseText';
import withTheme from '../../Theme/withTheme';

const styles = StyleSheet.create({
  bodyText: {
    fontSize: 16,
    lineHeight: 16,
    letterSpacing: 0.5,
  },
  bodyTextTwo: {
    fontSize: 14,
    lineHeight: 14,
    letterSpacing: 0.25,
  },
});

class BodyText extends Component {
  static propTypes = {
    type: PropTypes.number,
    children: PropTypes.node,
  };

  render() {
    const { type } = this.props;
    const textStyle = type == 2 ? styles.bodyTextTwo : styles.bodyText;

    return (
      <BaseText typographyStyles={textStyle} {...this.props}>
        {this.props.children}
      </BaseText>
    );
  }
}

export default withTheme(BodyText);
