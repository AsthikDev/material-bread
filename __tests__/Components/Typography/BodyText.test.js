import React from 'react';
import BodyText from '../../../src/Components/Typography/BodyText';

import renderer from 'react-test-renderer';

test('it renders BodyText One', () => {
  const tree = renderer.create(<BodyText type={1}>BodyText</BodyText>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('it renders BodyText Two', () => {
  const tree = renderer.create(<BodyText type={2}>BodyText</BodyText>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('it renders BodyText One when a value isnt 1 or 2', () => {
  const tree = renderer.create(<BodyText type={3}>BodyText</BodyText>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('it renders BodyText One with other props', () => {
  const tree = renderer
    .create(
      <BodyText type={3} align="center" gutterBottom={true}>
        BodyText
      </BodyText>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
