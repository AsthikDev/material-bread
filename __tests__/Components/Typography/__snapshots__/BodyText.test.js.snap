// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders BodyText One 1`] = `
<Text
  style={
    Object {
      "color": "rgba(0, 0, 0, 0.87)",
      "fontSize": 16,
      "letterSpacing": 0.5,
      "lineHeight": 16,
    }
  }
  type={1}
  typographyStyles={
    Object {
      "fontSize": 16,
      "letterSpacing": 0.5,
      "lineHeight": 16,
    }
  }
>
  BodyText
</Text>
`;

exports[`it renders BodyText One when a value isnt 1 or 2 1`] = `
<Text
  style={
    Object {
      "color": "rgba(0, 0, 0, 0.87)",
      "fontSize": 16,
      "letterSpacing": 0.5,
      "lineHeight": 16,
    }
  }
  type={3}
  typographyStyles={
    Object {
      "fontSize": 16,
      "letterSpacing": 0.5,
      "lineHeight": 16,
    }
  }
>
  BodyText
</Text>
`;

exports[`it renders BodyText Two 1`] = `
<Text
  style={
    Object {
      "color": "rgba(0, 0, 0, 0.87)",
      "fontSize": 14,
      "letterSpacing": 0.25,
      "lineHeight": 14,
    }
  }
  type={2}
  typographyStyles={
    Object {
      "fontSize": 14,
      "letterSpacing": 0.25,
      "lineHeight": 14,
    }
  }
>
  BodyText
</Text>
`;
